pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -526,7 +529,7 @@ func (t *tokenProvider) OAuthToken() string {
 
 func (a *Azure) authorizerForManagedIdentity() (autorest.Authorizer, error) {
 	msiConfig := kvauth.NewMSIConfig()
-	msiConfig.Resource = vaultResource
+	msiConfig.Resource = kvResourceForProviderConfig(a.provider)
```

Comment:
We don't need to provide the aadEndpoint, MSIConfig uses a multicast address `169.254.169.254` to talk to a internal auth API.